[GTK] Remove internal nested wayland compositor

carlosgcampos opened the pull request:
#### cfe6c2f6af2a215de0e5eb574c4c03521b6cce6f
<pre>
[GTK] Remove internal nested wayland compositor
<a href="https://bugs.webkit.org/show_bug.cgi?id=245873">https://bugs.webkit.org/show_bug.cgi?id=245873</a>

Reviewed by Michael Catanzaro and Adrian Perez de Castro.

WPE is currently available in all the distros we currently support, so
we can make wpe renderer mandatory and remove the old nested wayland
compositor.

* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.cpp:
(WebCore::PlatformDisplayWayland::initialize):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.h:
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::forceRepaint):
* Source/WebKit/Shared/gtk/WebKitWaylandProtocol.xml: Removed.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
(WebKit::bindWayland):
(WebKit::bubblewrapSpawn):
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStore.cpp:
(WebKit::AcceleratedBackingStore::checkRequirements):
(WebKit::AcceleratedBackingStore::create):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp:
(WebKit::tryInitializeSHM):
(WebKit::AcceleratedBackingStoreWayland::checkRequirements):
(WebKit::AcceleratedBackingStoreWayland::AcceleratedBackingStoreWayland):
(WebKit::AcceleratedBackingStoreWayland::~AcceleratedBackingStoreWayland):
(WebKit::AcceleratedBackingStoreWayland::unrealize):
(WebKit::AcceleratedBackingStoreWayland::tryEnsureTexture):
(WebKit::AcceleratedBackingStoreWayland::snapshot):
(WebKit::AcceleratedBackingStoreWayland::paint):
(WebKit::AcceleratedBackingStoreWayland::realize): Deleted.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.h:
* Source/WebKit/UIProcess/gtk/WaylandCompositor.cpp: Removed.
* Source/WebKit/UIProcess/gtk/WaylandCompositor.h: Removed.
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::updatePreferences):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
* Source/cmake/OptionsGTK.cmake:

Canonical link: <a href="https://commits.webkit.org/255081@main">https://commits.webkit.org/255081@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b78f541d1194b8283b4e0dcf99f09b85e5910a44

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91165 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/165 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21715 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/100715 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/160289 "Reverted pull request changes (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95171 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/192 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29197 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83527 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97282 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96821 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/153 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77915 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27119 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82100 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81726 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70152 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/82637 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35312 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15757 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/77661 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33108 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16778 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26772 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3528 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36892 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39682 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/80261 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38819 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35863 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17597 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -385,6 +370,20 @@ if (USE_OPENGL_OR_ES)
         SET_AND_EXPOSE_TO_BUILD(USE_GLX TRUE)
     endif ()
 
+    if (ENABLE_WAYLAND_TARGET AND EGL_FOUND)
```
What happens if EGL is not found...? Is it possible to wind up here by mistake, e.g. if the EGL pkg-config file is not installed? The build should fail if a build dep is missing, not proceed with the feature disabled.

carlosgcampos replied:
EGL is required for ENABLE_WAYLAND_TARGET and we make that fatal, but it happens after this, so I just added the AND to avoid doing it when configure is going to fail. Maybe we could move the EGL check before this one, but that's out of scope of this patch, I think.

aperezdc reviewed (approved):
r=me, always nice to see code being deleted 💣

carlosgcampos committed:
[GTK] Remove internal nested wayland compositor
https://bugs.webkit.org/show_bug.cgi?id=245873

Reviewed by Michael Catanzaro and Adrian Perez de Castro.

WPE is currently available in all the distros we currently support, so
we can make wpe renderer mandatory and remove the old nested wayland
compositor.

* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.cpp:
(WebCore::PlatformDisplayWayland::initialize):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.h:
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::forceRepaint):
* Source/WebKit/Shared/gtk/WebKitWaylandProtocol.xml: Removed.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
(WebKit::bindWayland):
(WebKit::bubblewrapSpawn):
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStore.cpp:
(WebKit::AcceleratedBackingStore::checkRequirements):
(WebKit::AcceleratedBackingStore::create):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp:
(WebKit::tryInitializeSHM):
(WebKit::AcceleratedBackingStoreWayland::checkRequirements):
(WebKit::AcceleratedBackingStoreWayland::AcceleratedBackingStoreWayland):
(WebKit::AcceleratedBackingStoreWayland::~AcceleratedBackingStoreWayland):
(WebKit::AcceleratedBackingStoreWayland::unrealize):
(WebKit::AcceleratedBackingStoreWayland::tryEnsureTexture):
(WebKit::AcceleratedBackingStoreWayland::snapshot):
(WebKit::AcceleratedBackingStoreWayland::paint):
(WebKit::AcceleratedBackingStoreWayland::realize): Deleted.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.h:
* Source/WebKit/UIProcess/gtk/WaylandCompositor.cpp: Removed.
* Source/WebKit/UIProcess/gtk/WaylandCompositor.h: Removed.
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::updatePreferences):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
* Source/cmake/OptionsGTK.cmake:

Canonical link: https://commits.webkit.org/255081@main
